models/sudoswap/ethereum/sudoswap_ethereum_pools.sql

```diff
@@ -0,0 +1,253 @@
+{{
+ config(
+       alias='pools',
+       post_hook='{{ expose_spells(\'["ethereum"]\',
+                                   "project",
+                                   "sudoswap",
+                                   \'["niftytable"]\') }}')
+}}
+
+WITH
+  pairs_created AS (
+    SELECT
+      CASE
+        WHEN _bondingCurve = '0x5b6ac51d9b1cede0068a1b26533cace807f883ee' THEN 'linear_bonding'
+        ELSE 'exp_bonding'
+      END as pricing_type,
+      _delta / 1e18 as delta,
+      CASE
+        WHEN _poolType = 0 THEN 'token'
+        WHEN _poolType = 1 THEN 'nft'
+        WHEN _poolType = 2 THEN 'trade'
+      END AS pooltype,
+      _spotPrice / 1e18 AS spotprice,
+      _nft AS nftcontractaddress,
+      _initialNFTIDs AS nft_ids,
+      _fee AS initialfee,
+      output_pair AS pair_address,
+      call_block_time AS block_time,
+      tx.FROM AS creator_address,
+      date_trunc('day', now()) - date_trunc('day', call_block_time) AS days_passed,
+      cardinality(_initialNFTIDs) AS initial_nft_count,
+      tx.value / 1e18 AS initial_eth
+    FROM
+      {{ source('sudo_amm_ethereum','LSSVMPairFactory_call_createPairETH') }} cre
+      INNER JOIN {{ source('ethereum','transactions') }} tx ON tx.hash = cre.call_tx_hash
```

Comment:
this model will be materialized as a view (based on dbt_project yaml file and no override in config block in this model). with this being a view, a join on a base table like ethereum transactions on only tx_hash is not efficient on spark and will take a long time to query each execution.

do you think this should be materialized as a delta table and incrementally update?

Reply:
I'll try to make it a delta table with increments instead based on the example I see in sudoswap_ethereum_events

Reply:
First time, doing one of these delta tables. In the config, would it be fine to only use the pool address as the 'unique_key'? Is using the block_date a must?

Reply:
```block_date``` is useful in the ```partition_by``` property, as it can help downstream query performance if you use date filter on dashboards. it's similar to how base tables (like ```ethereum.transactions```) is partitioned by block_date. 

it's not necessarily needed for ```unique_key``` -- the question to answer for this property is "what fields in my target table make it unique, so when i perform a merge statement, it will properly insert new and update existing?"

Reply:
I see, thanks for the clarification. I used `block_date` in `partition_by`. 

I used ['pool_address', 'nft_contract_address', 'creator_address'] for my unique key since the final table is pool based instead of time based. So new pools will be added and existing pools will be updated. Hope that was the right logic 😅

Reply:
Ah, my test failed because I dont have `block_date` in my final table. Thought it was an internal variable, didnt realise it needed to be in there.

Would it work if I partitioned by the date of the pools creation? 
Since its not an event based table there isnt really other date related variables there

Reply:
yeah, definitely partition by whatever makes the most sense for your target object. the goal should be a column which is frequently used in filters and/or joins downstream

Reply:
pool contract address is the most frequently used to filter and join. Will try using that to partition by